### PR TITLE
ST: refactor and improve TracingST

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

_Select the type of your PR_

- Bugfix
- Refactoring

### Description

This PR deals with several things connected to `TracingST`:
- adding wait for jaeger operator and instance deployment
- use install files for jaeger operator from upstream examples instead of our copy
- reuse one kafka cluster and jaeger operator for all tests
- move tracing clients from deployment to job
- remove internal clients from `TracingST`

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java [hunk 1]
```diff
@@ -187,101 +140,69 @@ void testConnectService() {
...
-            .withListenerName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
-            .build();
+        KafkaConnectorResource.createAndWaitForReadiness(KafkaConnectorResource.kafkaConnector(CLUSTER_NAME)
+            .editSpec()
+                .withClassName("org.apache.kafka.connect.file.FileStreamSinkConnector")
+                .addToConfig("file", Constants.DEFAULT_SINK_FILE_PATH)
+                .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .addToConfig("topics", topicName)
+                .endSpec()
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java:183]: Wrong indent

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java [hunk 2]
```diff
@@ -650,40 +467,40 @@ void testProducerConsumerMirrorMakerConnectStreamsService() {
         // TODO issue #4152 - temporarily disabled for Namespace RBAC scoped
         assumeFalse(Environment.isNamespaceRbacScope());
 
-        final String kafkaClusterSourceName = clusterName + "-source";
-        final String kafkaClusterTargetName = clusterName + "-target";
+        final String kafkaClusterSourceName = CLUSTER_NAME;
+        final String kafkaClusterTargetName = CLUSTER_NAME + "-target";
 
-        KafkaResource.createAndWaitForReadiness(KafkaResource.kafkaEphemeral(kafkaClusterSourceName, 3, 1).build());
+//        KafkaResource.createAndWaitForReadiness(KafkaResource.kafkaEphemeral(kafkaClusterSourceName, 3, 1).build());
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java:473]: Should this be commented?

**Review by im-konge — COMMENTED** [review]
Just nits, LGTM otherwise.

**Review by im-konge — APPROVED** [review]

**Pull request merged**